src/wireviz/Harness.py

```diff
@@ -330,90 +351,118 @@ def output(self, filename: (str, Path), view: bool = False, cleanup: bool = True
-                for index, color in enumerate(bundle.colors, 0):
-                    wirelist.append({'type': bundle.type, 'gauge': bundle.gauge, 'gauge_unit': bundle.gauge_unit, 'length': bundle.length, 'color': color, 'designator': bundle.name,
-                                     'manufacturer': remove_line_breaks(index_if_list(bundle.manufacturer, index)),
-                                     'mpn': remove_line_breaks(index_if_list(bundle.mpn, index)),
-                                     'pn': index_if_list(bundle.pn, index)})
-        # join similar wires from all the bundles to a single BOM item
-        wire_group = lambda w: (w.get('type', None), w['gauge'], w['gauge_unit'], w['color'], w['manufacturer'], w['mpn'], w['pn'])
-        for group in Counter([wire_group(v) for v in wirelist]):
-            items = [v for v in wirelist if wire_group(v) == group]
+        for cable in self.cables.values():
+            if not cable.ignore_in_bom:
+                # create name beilds used by both cables and bundles
+                cable_type = f', {remove_line_breaks(cable.type)}' if cable.type else ''
+                gauge_name = f' x {cable.gauge} {cable.gauge_unit}' if cable.gauge else ' wires'
+                if cable.category != 'bundle':
+                    # process cable as a single entity
+                    shield_name = ' shielded' if cable.shield else ''
+                    name = f'Cable{cable_type}, {cable.wirecount}{gauge_name}{shield_name}'
+                    item = {'item': name, 'qty': cable.length, 'unit': 'm', 'designators': cable.name,
+                            'manufacturer': remove_line_breaks(cable.manufacturer), 'mpn': remove_line_breaks(cable.mpn), 'pn': cable.pn}
+                    bom_items.append(item)
+                else:
+                    # add each wire from the bundle to the bom
+                    for index, color in enumerate(cable.colors, 0):
+                        wire_color = f', {color}' if color else ''
+                        name = f'Wire{cable_type}{gauge_name}{wire_color}'
+                        item = {'item': name, 'qty': cable.length, 'unit': 'm', 'designators': cable.name,
+                                'manufacturer': remove_line_breaks(index_if_list(cable.manufacturer, index)),
+                                'mpn': remove_line_breaks(index_if_list(cable.mpn, index)), 'pn': index_if_list(cable.pn, index)}
+                        bom_items.append(item)
+
+            for part in cable.additional_components:
+                qty = part.qty * cable.get_qty_multiplier(part.qty_multiplier)
+                bom_items.append(
+                    {
+                        'item': part.description,
+                        'qty': qty,
+                        'unit': part.unit,
+                        'manufacturer': part.manufacturer,
+                        'mpn': part.mpn,
+                        'pn': part.pn,
+                        'designators': cable.name
```

Comment:
I guess this code line should have ended with `if cable.show_name else None` as for connectors, but by using a common method, this should be fixed.